src/response.rs

```diff
@@ -507,18 +562,44 @@ impl Response {
 
         let url = unit.url.clone();
 
-        Ok(Response {
+        // Prefetch data if it's small enough to be buffered in one go.
+        let peeked_len = stream.fill_buf()?.len();
```

Comment:
I think we shouldn't call `fill_buf()` here. It [was already called by BufReader at least once](https://doc.rust-lang.org/nightly/src/std/io/buffered/bufreader.rs.html#257-369), the moment we read from this response. Calling it again will result in another syscall, even in the common case where we've already read the full body but just don't know it yet because we haven't checked.

Also I think `peeked_len` should be called `bytes_available` or similar, since we haven't already peeked.

Reply:
So how do we know whether to buffer the response?

Reply:
Ok. I'll do that.